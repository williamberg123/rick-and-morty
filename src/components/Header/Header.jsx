import React, { memo } from 'react';

import './Header.css';

function Header() {    
    return (
        <header className="Header">
            <span>Rick & Morty</span>
        </header>
    );
}

export default memo(Header);