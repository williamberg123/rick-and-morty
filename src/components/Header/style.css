.Header {
    display: flex;
    align-items: center;
    justify-content: flex-start;
    padding: 20px;
    background-color: rgb(0, 71, 129);
    color: white;
    font-weight: bold;
}

@media (max-width: 450px) {
    .Header {
        padding: 10px;
    }
}