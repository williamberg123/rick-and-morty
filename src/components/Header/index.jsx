import React, { memo } from 'react';

import './style.css';

function Header() {    
    return (
        <header className="Header">
            <span>Rick & Morty</span>
        </header>
    );
}

export default memo(Header);