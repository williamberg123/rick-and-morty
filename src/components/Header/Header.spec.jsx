import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from './Header';

describe('<Header />', () => {
    it('should render span with the text', () => {
        render(<Header />);

        expect(screen.getByText(/rick & morty/i)).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        render(<Header />);
        expect(screen.getByText(/rick & morty/i).parentElement).toMatchSnapshot();
    });
});

// Testado