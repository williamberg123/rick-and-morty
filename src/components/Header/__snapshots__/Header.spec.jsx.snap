// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<header
  class="Header"
>
  <span>
    Rick & Morty
  </span>
</header>
`;
